cmake/app/boilerplate.cmake

```diff
@@ -535,6 +540,34 @@ if(CONFIG_QEMU_TARGET)
   endif()
 endif()
 
+# This needs to be before adding ZEPHYR_BASE subdirectory
+# sice PARTITION_MANAGER_CONFIG_FILES is checked there.
+if(EXISTS ${APPLICATION_SOURCE_DIR}/pm.yml)
+  execute_process(
+    COMMAND ${CMAKE_C_COMPILER}
+    -x assembler-with-cpp
+    -nostdinc
+    -I${PROJECT_BINARY_DIR}/include/generated
+    ${NOSYSDEF_CFLAG}
+    -P
+    -E ${APPLICATION_SOURCE_DIR}/pm.yml
+    -o ${PROJECT_BINARY_DIR}/include/generated/pm.yml
+    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+    RESULT_VARIABLE ret
+    )
+  if(NOT "${ret}" STREQUAL "0")
+    message(FATAL_ERROR "command failed with return code: ${ret}")
+  endif()
+  string(LENGTH ${IMAGE} len)
+  MATH(EXPR len "${len}-1")
+  string(SUBSTRING ${IMAGE} 0 ${len} image_name)
+  set_property(
+    GLOBAL APPEND PROPERTY
+    PARTITION_MANAGER_CONFIG_FILES
+    "${image_name}:${PROJECT_BINARY_DIR}/include/generated/pm.yml:${PROJECT_BINARY_DIR}:${PROJECT_BINARY_DIR}/include/generated"
+    )
+endif()
+
```

Comment:
This hunk should be moved to nrf/subsys/partition_manager

Reply:
moved